### PR TITLE
Add missing lock in submitblock

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Also for 0.10

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
This just fixes the bug, it would be nice to use threadsafetey.h annotations, but I believe that requires build-system changes as well.

File: src/rpcmining.cpp [hunk 1]
```diff
@@ -629,14 +629,19 @@ Value submitblock(const Array& params, bool fHelp)
...
     uint256 hash = block.GetHash();
-    BlockMap::iterator mi = mapBlockIndex.find(hash);
-    if (mi != mapBlockIndex.end()) {
-        CBlockIndex *pindex = mi->second;
-        if (pindex->IsValid(BLOCK_VALID_SCRIPTS))
-            return "duplicate";
-        if (pindex->nStatus & BLOCK_FAILED_MASK)
-            return "duplicate-invalid";
-        // Otherwise, we might only have the header - process the block before returning
+    bool fBlockPresent;
```
Comment (jtimon) [inline src/rpcmining.cpp]: Little nit: wouldn't be better to explicitly initialize this to false here ?

Comment (TheBlueMatt) [inline src/rpcmining.cpp]: ehhh, lol yea

**Comment by gmaxwell** [comment]
This looks good to me now. untested ACK.

**Comment by luke-jr** [comment]
utACK

**Comment by laanwj** [comment]
utACK

**Comment by laanwj** [comment]
Cherry-picked to 0.10 as eae305f

**Pull request merged**